trio/_core/_multierror.py

```diff
@@ -126,9 +126,16 @@ def __enter__(self):
         pass
 
     def __exit__(self, etype, exc, tb):
+
+        force_raise = False
+        if protector.has_unawaited_coroutines():
+            exc =  protector.make_non_awaited_coroutines_error(
+                protector.pop_all_unawaited_coroutines()
+            )
+            force_raise = True
         if exc is not None:
             filtered_exc = MultiError.filter(self._handler, exc)
-            if filtered_exc is exc:
+            if filtered_exc is exc and not force_raise:
```

Comment:
I think this would be clearer if rewritten like:
```python
original_exc = exc
if protector.has_unawaited_coroutines():
    exc = ...
if exc is not None:
    exc = MultiError.filter(...)
if exc is original_exc:
    ...
```